containers/front-end/src/app/heatmap/heatmap.component.ts

```diff
@@ -78,10 +91,19 @@ export class HeatmapComponent implements OnInit {
    * Creates a random step and colors grid cell every second
    */
   public colorHeatMap(): void {
-    this.heatMapInterval = setInterval(() => {
-      const step = randomStep(this.HEATMAP_ROWS - 1, this.HEATMAP_COLUMNS - 1);
-      this.changeGridCell(step['x'], step['y']);
-    }, 250);
+    const step = randomStep(this.HEATMAP_ROWS - 1, this.HEATMAP_COLUMNS - 1);
+    this.addDegradedCell(step);
+    this.changeGridCell(step['x'], step['y'], false);
+  }
+
+  /**
+   * Creates a random step and colors grid cell every second
+   */
+  public changeDegradedCell(): void {
+    const step = this.retrieveDegradedCell();
```

Comment:
Add an if statement to check if degradedcells array length > 500
This allows a sliding window of the latest 500 heat map grid points at a time